plugins/main/bridge/bridge.go

```diff
@@ -677,39 +676,18 @@ func cmdAdd(args *skel.CmdArgs) error {
 			}
 		}
 	} else {
+		// If layer 2 we still need to set the container veth to up
 		if err := netns.Do(func(_ ns.NetNS) error {
-			link, err := netlink.LinkByName(args.IfName)
```

Comment:
Why do you want a function for this? I understand the one for retries which includes some logic with the for loop but this is simple and not used anywhere else.

Reply:
My intent was to improve readability, even though there is no much of logic its less code to read, and may reduce some cognitive load (I dont need to know we need to fetch the link in order to bring it down).

Also, with #997 changes, it makes the code read better: 
``` 
...
else if !disableContIface {
        link.SetUp(ifaceName)
}
...
```
WDYT?

Reply:
no, I think this is simple enough. If you want to improve readability, then abstract all the logic inside if layer3 branch, then in your other pr, you can do
`
if !disableContIface {
    if layer {
        layer3func()
    } else {
         thiscode
    }
    WaitOperStateUp()
}
`